Frameworks/Foundation/NSThread.mm

```diff
@@ -356,8 +356,15 @@ + (NSThread*)currentThread {
     NSThread* currentThread = [[self class] _threadObjectFromCurrentThread];
 
     if (currentThread == nil) {
-        currentThread = [NSThread new];
+        // Underlying thread created outside the NSThread APIs
+
+        // So create a new instance
+        currentThread = [[NSThread alloc] init];
         [currentThread _associateWithCurrentThread];
```

Comment:
Good find and somewhat worrisome.  Yes, with NSProgress.mm, there is a leak.  s_currentProgressStack could become a thread local static inside _getProgressStackForCurrentThread much like you have done here.